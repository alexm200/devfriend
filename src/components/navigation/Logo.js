import React, { Component } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

class Logo extends Component {
  render() {
    return (
        <a className="sidebar-brand d-flex align-items-center justify-content-center" href="/">
            <div className="sidebar-brand-icon">
                <FontAwesomeIcon icon="laptop-code" />                    
            </div>
            <div className="sidebar-brand-text mx-3">Dev Friend</div>
        </a>
    );
  }
}

export default Logo;