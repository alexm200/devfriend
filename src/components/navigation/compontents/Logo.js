import React, { Component } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import styles from '../../../styles/modules/navigation.module.scss';
import { NavLink } from "react-router-dom";

class Logo extends Component {
  render() {
    return (
        <NavLink className={[styles.brand, "d-flex", "align-items-center", "justify-content-center"].join(" ")} to="/">
            <div className={styles.icon}>
                <FontAwesomeIcon icon="laptop-code" />                    
            </div>
            <div className={[styles.text, "mx-3"].join(" ")}>Dev Friend</div>
        </NavLink>
    );
  }
}

export default Logo;