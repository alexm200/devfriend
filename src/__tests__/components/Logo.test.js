import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import Logo from '../../components/navigation/components/Logo';

describe('<Logo />', () => {
    describe('render()', () => {
      test('renders the component', () => {
        const wrapper = shallow(<Logo />);
        const component = wrapper.dive();
  
        expect(toJson(component)).toMatchSnapshot();
      });
    });
  });