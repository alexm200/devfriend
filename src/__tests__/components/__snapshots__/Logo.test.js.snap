// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> render() renders the component 1`] = `
<Route
  path="\\\\/"
>
  <Component />
</Route>
`;
